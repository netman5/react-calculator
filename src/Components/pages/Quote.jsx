import React from 'react';

function Quote() {
  return (
    <>
      <div className="quote" data-testid="ancestor">
        <blockquote data-testid="descendant">
          Mathematics is not about numbers, eqautions, computations or algorithms;
          it is about understanding. â€”William Paul Thurston
        </blockquote>
      </div>
    </>
  );
}

export default Quote;
