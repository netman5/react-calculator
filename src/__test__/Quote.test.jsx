import renderer from 'react-test-renderer';
import { getByTestId, render, screen } from '@testing-library/react';
import Quote from '../Components/pages/Quote';

describe('Test component using Jest', () => {
  it('Test Quote component if it matches render', () => {
    const component = renderer.create(<Quote />).toTree();

    expect(component).toMatchSnapshot();
  });
});
