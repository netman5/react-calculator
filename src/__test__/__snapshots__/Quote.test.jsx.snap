// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test component using Jest Test Quote component if it matches render 1`] = `
Object {
  "instance": null,
  "nodeType": "component",
  "props": Object {},
  "rendered": Object {
    "instance": null,
    "nodeType": "host",
    "props": Object {
      "children": <blockquote
        data-testid="descendant"
      >
        Mathematics is not about numbers, eqautions, computations or algorithms; it is about understanding. —William Paul Thurston
      </blockquote>,
      "className": "quote",
      "data-testid": "ancestor",
    },
    "rendered": Array [
      Object {
        "instance": null,
        "nodeType": "host",
        "props": Object {
          "children": "Mathematics is not about numbers, eqautions, computations or algorithms; it is about understanding. —William Paul Thurston",
          "data-testid": "descendant",
        },
        "rendered": Array [
          "Mathematics is not about numbers, eqautions, computations or algorithms; it is about understanding. —William Paul Thurston",
        ],
        "type": "blockquote",
      },
    ],
    "type": "div",
  },
  "type": [Function],
}
`;
